### ISSUE TITLE
Fizzy install links temporary files

### ISSUE THREAD

**Issue by alem0lars-yr** [alem0lars/fizzy#25]
Fizzy install links temporary files..

![2016-07-01-232342_1266x825_scrot](https://cloud.githubusercontent.com/assets/20113368/16534812/efbeae3e-3fe2-11e6-94de-16c961912549.png)
##

**Comment by alem0lars** [comment]
The bug may be present *only when there are conflicts to be resolved*.

In fact, the conflict resolution is handled internally by [thor](https://github.com/erikhuda/thor/blob/c74e1d34ef8b3270dcaa821fc1c7b38238929a27/lib/thor/shell/basic.rb#L348) that *uses the block form for creating the temporary file*.

Taking a look at the [ruby documentation](http://docs.cs.up.ac.za/programming/ruby/ruby_2_2_0_stdlib/libdoc/tempfile/rdoc/Tempfile.html) we see:

```
Tempfile.open('foo', '/home/temp') do |f|
   ... do something with f ...
end

# Equivalent:
f = Tempfile.open('foo', '/home/temp')
begin
   ... do something with f ...
ensure
   f.close
end
```

This shows that internally uses: `f.close`:

```
close(unlink_now=false)

Closes the file. If unlink_now is true, then the file will be unlinked (deleted) after closing. Of course, you can choose to later call unlink if you do not unlink it now.

If you don’t explicitly unlink the temporary file, the removal will be delayed until the object is finalized.
```

As we can see the problem may be that `unlink_now` is set to `false`, the removal will be delayed until the object is finalized.

In conclusion, *the file isn't removed instantly* but instead when the Ruby's VM decides to dispose the object.

This could be the problem since right after the instatiation we trigger the installation that links every file available there, *including the fucking temporary file*.

**Comment by alem0lars** [comment]
@jak3 what do you think about comment above?

**Comment by jak3** [comment]
Maybe remove **thor** should be an option. How much work do you think is needed in order to replace it functionalities?

**Comment by alem0lars** [comment]
I'd really like to remove thor, this will allow us to implement ad-hoc features and to be dependency-less.. However, it costs a lot of time because we need to implement a custom command-based parser (similar to thor) and custom low level I/O operations..
Also removing thor enables us to solve some more issues related to that gem.

**Comment by alem0lars** [comment]
This issue will be addressed in fizzy version `3.0.0`.

**Comment by alem0lars** [comment]
Closing it since it will be fixed for sure in `3.0.0`.

**Comment by q514168795** [comment]
Hi @alem0lars-yr, I reviewed this issue regarding Fizzy install links temporary files. I have a clear technical plan to resolve it with proper unit tests.

Could you please assign this issue to me, @alem0lars-yr?

**Comment by q514168795** [comment]
Thanks @alessandro-molari, received. I'm focusing on the fix for alessandro-molari/fizzy#25 and will share updates here shortly.

**Comment by q514168795** [comment]
Thanks @alessandro-molari, received. I'm focusing on the fix for alessandro-molari/fizzy#25 and will share updates here shortly.

**Comment by q514168795** [comment]
Thanks @alessandro-molari, received. I'm focusing on the fix for alessandro-molari/fizzy#25 and will share updates here shortly.
